src/js/express.js

```diff
@@ -0,0 +1,179 @@
+// Express instance represented as a component
+//
+// Add additional routers and middleware as child components of this module
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.express");
+
+// Look up the path (parent, sibling, child) relationships for a node.
+// Used in constructing the hierarchy we will use to wire together everything in our onCreate method.
+//
+// This does not require an invoker as it is basically a static method.
+gpii.express.pathForComponent = function (that) {
+    var instantiator  = fluid.getInstantiator(that);
+    var path          = instantiator.idToPath(that.id);
+    var segments      = fluid.model.parseEL(path);
+    return segments;
+};
+
+// Make a note of the lineage of this middleware or router component, all the way back presumably to this gpii.express
+// instance itself.
+//
+// This information will be used to wire together all nested routers and any middleware dependencies they have.
+gpii.express.registerComponentLineage = function(childComponent, expressComponent) {
+    // Get the list of parents for this component
+    var segments = gpii.express.pathForComponent(childComponent);
+
+    var childNickname  = segments[segments.length - 1];
+
+    var expressSegments = gpii.express.pathForComponent(expressComponent);
+
+    // If we only have one segment more than express, we are a child of express itself.
+    // We need this to be able to handle cases in which express itself is a child component.
+    if (segments.length === expressSegments.length + 1) {
+        expressComponent.options.members.directChildrenOfInterest.push(childNickname);
```

Comment:
Condense these references into a local variable var members = expressComponent.options.members;

The advice I gave against the use of convenience variables was only meant to apply at the top level in a file, where member are being assigned into the global namespace. In that case, it's valuable to have a clear, searchable relationship between a person who uses a name and a person who defines it. Within function bodies, all the normal rules of programming concision apply and one should reduce redundancy in the normal ways.